src/raftstore/store/worker/region.rs

```diff
@@ -161,6 +270,9 @@ impl SnapContext {
         let start_key = keys::enc_start_key(&region);
         let end_key = keys::enc_end_key(&region);
         check_abort(&abort)?;
+        if self.get_delay_secs() > 0 {
+            self.destroy_overlap_ranges(&start_key, &end_key);
+        }
```

Comment:
Should have an `else` here?

Reply:
No, if `delay` is 0, we delete the range as soon as it comes in.